test/packet: extend add_vagrant_box.sh script to load dev/CI images

qmonnet opened the pull request:
Extend the script used to preload Ubuntu images for the CI with packet, so it can be used to preload images for tests and dev in general.

Changes include:

- Make `vagrant_box_defaults.rb` path optional (defaults to `./vagrant_box_defaults.rb`).
- Add a `-b` option to pass the names of the boxes to download.
- Add a `-l` option to download latest available versions for the images instead of version numbers from `vagrant_box_defaults.rb`.
- Add a `-a` option to download with `aria2c` instead of `curl`.
- Add a `-t` option to download the images under `/tmp/` instead of using the current directory.
- Extend help message to reflect these changes and provide a couple of example invocations.
  <details><summary>New help message</summary>

  ```
  usage: add-vagrant-boxes.sh [options] [vagrant_box_defaults.rb path]
          path to vagrant_box_defaults.rb defaults to ./vagrant_box_defaults.rb
  
  options:
          -a              use aria2c instead of curl
          -b <box>        download selected box (defaults: ubuntu ubuntu-next)
          -l              download latest versions instead of using vagrant_box_defaults
          -t              download to /tmp/ instead of current directory
          -h              display this help
  
  examples:
          download boxes ubuntu and ubuntu-next from vagrant_box_defaults.rb:
          $ add-vagrant-boxes.sh $HOME/go/src/github.com/cilium/cilium/vagrant_box_defaults.rb
          download latest version for ubuntu-dev and ubuntu-next:
          $ add-vagrant-boxes.sh -l -b ubuntu-dev -b ubuntu-next
          same as above, downloading into /tmp/ and using aria2c:
          $ add-vagrant-boxes.sh -alt -b ubuntu-dev -b ubuntu-next
  ```
  </details>

The behaviour of the script is unchanged if invoked with none of those new options.

Currently marking as a RFC because I don't know if there is any interest for extending that script, and I can keep a local version if I'm the only one finding it useful. If others were to find it helpful, a follow-up could be to move that script under `contrib/` and to mention it in the docs.

pchaigno commented on test/packet/scripts/add_vagrant_box.sh:
```diff
@@ -1,55 +1,169 @@
+    echo -e ""
+    echo -e "options:"
+    echo -e "\t-a\t\tuse aria2c instead of curl"
+    echo -e "\t-b <box>\tdownload selected box (defaults: ubuntu ubuntu-next)"
```
Maybe it should default to all, so `ubuntu-dev`, `ubuntu-next`, and `ubuntu` if I recall correctly.

pchaigno replied:
Ah, I guess you defaulted it this way to avoid changing the current behavior?

pchaigno commented on test/packet/scripts/add_vagrant_box.sh:
```diff
@@ -1,55 +1,169 @@
+        if [[ $use_aria2 -eq 1 ]] ; then
+            aria2c -x16 -s16 -c -d "$outdir" -o package.box "$url"
+            vagrant box add "cilium/$box" "$outdir/package.box"
+            mv $box_dir$box/{0,$version}
```
The `metadata_url` file also needs to be created here, for the first VM downloaded. Otherwise, Vagrant complains when trying to start the VM.

qmonnet replied:
Ah right, thanks. I'll add it.

aanm replied:
Just to be clear, it needs to be created in the `$box_dir$box/`

pchaigno reviewed (approved):
This is awesome! I'll definitely add to cron :-)

nebril commented on test/packet/scripts/add_vagrant_box.sh:
```diff
@@ -1,55 +1,174 @@
+check_cmd curl vagrant
+
+usage() {
+    echo -e "usage: add-vagrant-boxes.sh [options] [vagrant_box_defaults.rb path]"
```
`s/add-vagrant-boxes/add-vagrant-box` (error left over from old comment).

qmonnet replied:
Thanks I'll fix that.

nebril reviewed (commented):
This is the first time I see these scripts, I never used them. They are not used in CI, although similar ones live in our internal infra repository, so it's ok to merge them, although I think they should be moved to `contrib` dir.

qmonnet committed:
test/packet: extend add_vagrant_box.sh script

Extend the script used to preload Ubuntu images for the CI with packet,
so it can be used to preload images for tests and dev in general.

Changes include:

- Make vagrant_box_defaults.rb path optional (defaults to
  ./vagrant_box_defaults.rb).

- Add a -b option to pass the names of the boxes to download.

- Add a -l option to download latest available versions for the images
  instead of version numbers from vagrant_box_defaults.rb.

- Add a -a option to download with aria2c instead of curl.

- Add a -t option to download the images under /tmp/ instead of using
  the current directory.

- Extend help message to reflect these changes and provide a couple of
  example invocations.

The behaviour of the script is unchanged if invoked with none of those
new options.

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

qmonnet committed:
test/packet: fix call to add_vagrant_box.sh script

When the add_vagrant_box.sh script was added, the install.sh script for
packet was updated to use it instead of the equivalent file from
terraform (if I understand correctly). The shell script is copied and
given execution permission, but the call to the script itself has not
been updated. Let's fix it.

Fixes: #9850
Fixes: 0c2d13d31de9 ("test/packet: fix packet terraform scripts")
Signed-off-by: Quentin Monnet <quentin@isovalent.com>
